Inc/HALALMock/Services/PWM/DualPWM/DualPWM.hpp

```diff
@@ -1,27 +1,17 @@
-/*
- * DualPWMInstance.hpp
- *
- *  Created on: Feb 27, 2023
- *      Author: aleja
- */
-
 #pragma once
 
-#include "PWM/PWM/PWM.hpp"
+#include "HALALMock/Services/PWM/PWM/PWM.hpp"
 
 class DualPWM{
 protected:
 	DualPWM() = default;
-	TimerPeripheral* peripheral;
-	uint32_t channel;
-	float duty_cycle;
-	uint32_t frequency;
-	bool is_on = false;
-	static constexpr float CLOCK_FREQ_MHZ_WITHOUT_PRESCALER = 275;
-	static constexpr float clock_period_ns = (1/CLOCK_FREQ_MHZ_WITHOUT_PRESCALER)*1'000;
-	bool is_initialized = false;
+	float* duty_cycle{};
+	uint32_t* frequency{};
+	bool positive_is_on*{};
+	bool negative_is_on*{};
+	std::chrono::nanoseconds *dead_time_ns{};
 public:
-	DualPWM(Pin& pin, Pin& pin_negated);
+	DualPWM(Pin& pin_pos, Pin& pin_neg);
```

Comment:
why the change in the name?

Reply:
If you want to go with this I would call them `pin_positive` and `pin_negative` and give a new name to the EmulatedPins